### PR TITLE
Zq numtraits

### PR DESCRIPTION
**Pull request by 3pmile**
**Description**

This PR implements...
- [x] is_zero for Zq, MatZq, PolyOverZq
- [x] is_one for Zq, PolyOverZq
- [x] is_square for MatZq
- [x] is_identity for MatZq

as in https://github.com/qfall/math/pull/270

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

**Review by MoogSven — CHANGES_REQUESTED** [review]
Mostly the same comments as for #307.

File: src/integer_mod_q/mat_zq/properties.rs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    ///  and all other entries are `0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatZq::from_str("[[1, 0],[0, 1]] mod 17").unwrap();
+    /// assert!(value.is_identity())
+    /// ```
```
Comment (MoogSven) [inline src/integer_mod_q/mat_zq/properties.rs:30]: I like the test case you added for `MatQ::is_identity` to make it more clear that non-square matrices are allowed.
```suggestion
    /// ```
    ///
    /// ```
    /// use qfall_math::integer_mod_q::MatZq;
    /// use std::str::FromStr;
    ///
    /// let value = MatZq::from_str("[[1, 0],[0, 1], [0, 0]] mod 17").unwrap();
    /// assert!(value.is_identity());
    /// ```
```

File: src/integer_mod_q/mat_zq/properties.rs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    /// Returns true if every diagonal entry of the upper square matrix is `1`
+    ///  and all other entries are `0`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let value = MatZq::from_str("[[1, 0],[0, 1]] mod 17").unwrap();
+    /// assert!(value.is_identity())
```
Comment (MoogSven) [inline src/integer_mod_q/mat_zq/properties.rs:29]: Add a semicolon at the end of your assert macros. This applies to all doc tests.

**Review by MoogSven — APPROVED** [review]
Looks good, just small changes required. Check the comments for details.

File: src/integer_mod_q/mat_zq/properties.rs (hunk 2 above)
Comment (MoogSven) [reply src/integer_mod_q/mat_zq/properties.rs:29]: You can use `.*assert.*!(.*[^;]\n` in the regex search to find them.
